tsl/src/compression/compression.c

```diff
@@ -1962,8 +1963,18 @@ create_segment_filter_scankey(RowDecompressor *decompressor, char *segment_filte
 		elog(ERROR, "no btree opfamily for type \"%s\"", format_type_be(atttypid));
 
 	Oid opr = get_opfamily_member(tce->btree_opf, atttypid, atttypid, strategy);
-	Assert(OidIsValid(opr));
-	/* We should never end up here but: no operator, no optimization */
+
+	/*
+	 * Fall back to btree operator input type when it is binary compatible with
+	 * the column type and no operator for column type could be found.
+	 */
+	if (!OidIsValid(opr) && IsBinaryCoercible(atttypid, tce->btree_opintype))
+	{
+		opr =
```

Comment:
Codecov complains about missing coverage. However, the test case seems to cover this part of the code and removing these lines triggers a test failure. So, it seems to be tested.

Reply:
Codecov has been all over the place lately, not sure whats going on with it. Reporting random lines being untested which are in the same branch where all other lines have been tested.